diff --git a/scripts/build-docker.sh b/scripts/build-docker.sh
index 6f86fb46a..1f40a6d19 100755
--- a/scripts/build-docker.sh
+++ b/scripts/build-docker.sh
@@ -13,7 +13,7 @@ if [ -z "$VERSION" ]; then
   exit 1
 fi
 
-ARCH=$(go env GOARCH)
+ARCH=${ARCH:-$(go env GOARCH)}
 VERSION="${VERSION}-${ARCH}"
 DOCKERFILE="Dockerfile"
 
@@ -40,11 +40,17 @@ cp "${BINARYDIR}"/etcd "${BINARYDIR}"/etcdctl "${BINARYDIR}"/etcdutl "${IMAGEDIR
 
 cat ./"${DOCKERFILE}" > "${IMAGEDIR}"/Dockerfile
 
+BASE_IMAGE_ARG=""
+if [ "${ARCH}" = "loong64" ]; then
+    BASE_IMAGE=${BASE_IMAGE:-ghcr.io/loong64/debian:trixie-slim}
+    BASE_IMAGE_ARG="--build-arg=BASE_IMAGE=${BASE_IMAGE}"
+fi
+
 if [ -z "${TAG:-}" ]; then
     # Fix incorrect image "Architecture" using buildkit
     # From https://stackoverflow.com/q/72144329/
-    DOCKER_BUILDKIT=1 docker build --build-arg="ARCH=${ARCH}" -t "gcr.io/etcd-development/etcd:${VERSION}" "${IMAGEDIR}"
-    DOCKER_BUILDKIT=1 docker build --build-arg="ARCH=${ARCH}" -t "quay.io/coreos/etcd:${VERSION}" "${IMAGEDIR}"
+    DOCKER_BUILDKIT=1 docker build --build-arg="ARCH=${ARCH}" ${BASE_IMAGE_ARG} -t "gcr.io/etcd-development/etcd:${VERSION}" "${IMAGEDIR}"
+    DOCKER_BUILDKIT=1 docker build --build-arg="ARCH=${ARCH}" ${BASE_IMAGE_ARG} -t "quay.io/coreos/etcd:${VERSION}" "${IMAGEDIR}"
 else
-    docker build -t "${TAG}:${VERSION}" "${IMAGEDIR}"
+    docker build --build-arg="ARCH=${ARCH}" ${BASE_IMAGE_ARG} -t "${TAG}:${VERSION}" "${IMAGEDIR}"
 fi
